### ISSUE TITLE
browsergrad-jit load_state_dict strict=False still rejects missing keys

### ISSUE THREAD

**Issue by vijayksingh** [unlocalhosted/browsergrad#11]
## What happened

In both published browsergrad-jit 0.8.2 and current source, `Module.load_state_dict(state, strict=False)` skips the unexpected-key check but still raises `KeyError` for every expected Parameter or buffer absent from `state`.

## Expected

Match PyTorch `nn.Module.load_state_dict`: `strict=False` should permit missing and unexpected keys, still reject same-key shape mismatches, load compatible entries, and return incompatible-key evidence (`missing_keys`, `unexpected_keys`).

## Reproduction

```python
import torch

model = torch.nn.Linear(2, 3)
partial = {"weight": model.state_dict()["weight"]}
result = model.load_state_dict(partial, strict=False)
assert result.missing_keys == ["bias"]
assert result.unexpected_keys == []
```

BrowserGrad currently raises for the missing `bias`.

## Learner impact

Warm-start and partial-load exercises either teach false `strict=False` behavior or cannot execute the actual PyTorch contract. This also prevents checking the incompatible-key return value that practitioners should inspect.

## Scope

This is independent of shape mismatch behavior: a same-key incompatible shape should remain an error under both strict policies.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** bug
**Summary:** Make `load_state_dict(strict=False)` load partial checkpoints and report incompatible keys.

**Current behavior:**
A fresh Pyodide check removes `bias` from a Linear state dict and calls `load_state_dict(..., strict=False)`; it still raises a missing-key `KeyError`. The method returns `None` on success and only relaxes unexpected-key checking, so callers cannot inspect missing or unexpected keys.

**Desired behavior:**
Non-strict loading should copy every present, recognized, shape-compatible parameter and buffer, skip missing and unexpected keys, and return an incompatibility result containing deterministic `missing_keys` and `unexpected_keys` lists. Strict loading should reject either list. Shape incompatibility for a present recognized key must remain an error in both modes.

**Key interfaces:**
- `Module.load_state_dict(state, strict=True)` — return an `IncompatibleKeys`-style result and honor strictness for missing/unexpected keys.
- Nested parameter and buffer names — report fully qualified keys in stable order.
- State copy validation — preserve exact shape/dtype/session safety and pending-state synchronization.

**Acceptance criteria:**
- [ ] `strict=False` loads present compatible values and returns missing and unexpected key lists without throwing for those lists.
- [ ] `strict=True` rejects both missing and unexpected keys with a descriptive error.
- [ ] Present recognized keys with wrong shapes fail in both modes and do not silently corrupt state.
- [ ] Nested modules and BatchNorm parameters/buffers report fully qualified incompatible keys correctly.
- [ ] A complete compatible state dict returns empty lists and preserves current round-trip behavior.

**Out of scope:**
- Automatic key renaming or checkpoint migration.
- Silent cross-shape or unsupported dtype conversion.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Verified outcome

**State:** wontfix — already implemented

Implemented in https://github.com/unlocalhosted/browsergrad/commit/4d7181a9e14ba8c69e8714277da4172b361afcf4: load_state_dict(strict=False) loads compatible present values and returns deterministic missing/unexpected keys. Strict and shape failures validate atomically before mutation. Fresh verification passed JIT build, both typechecks, lint, 27 unit tests, 345 integration tests, architecture checks, packed-release tests, and the required JIT semantic-permute WebGPU gate.

Closing as completed and replacing the ready state with `wontfix` per the repository triage state machine.
